core/src/test/scala/io/projectglow/tertiary/LinearRegressionSuite.scala

```diff
@@ -304,6 +320,24 @@ class LinearRegressionSuite extends GlowBaseTest {
     assert(e.getCause.getMessage.contains(error))
   }
 
+  // Our linear regression algorithm projects the genotypes onto orthogonal complement of the
+  // covariate vector space. When the genotypes are a linear combination of some covariates, this
+  // projection is 0, so beta, standard error, and p value are all undefined. However, finite
+  // precision can instead cause the error to be massive with respect to beta.
+  test("negligible p value if genotypes are in covariate span") {
```

Comment:
Nit: this is a misleading name; negligible suggests the p-value is small when it's actually very large.